op-service/sources/prefetching_eth_client_test.go

```diff
@@ -0,0 +1,84 @@
+package sources
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/testutils"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+// TestPrefetchingEthClient runs all test cases for each prefetching range.
+func TestPrefetchingEthClient(t *testing.T) {
+	prefetchingRanges := []uint64{0, 1, 5}
+	for _, prefetchingRange := range prefetchingRanges {
```

Comment:
I suggest to run each test in a `t.Run` sub test.

Reply:
This also has the added benefit that we don't stack `defer` statements in a loop, which should generally be avoided.